packages/components/src/styles/components/dialog-primitive.scss

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// DIALOG PRIMITIVE
+//
+
+
+// MAIN CONTAINER (<dialog>)
+
+// :where reduces the specificity of the selector
+// & can be used to set “loose” defaults which can be easily overridden
+dialog:where(.hds-dialog-primitive) {
+  all: unset; // removes default dialog & inherited styles. Must be the first style to be applied
+  // unsetting all the styles causes it to not inherit the box-sizing from the parent
+  box-sizing: border-box;
+}
+
+
+// WRAPPER
+
+.hds-dialog-primitive__wrapper {
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  height: 100%;
+  background: var(--token-color-surface-primary);
+  border-radius: inherit;
+}
+
+// Note: this wraps both .hds-dialog-primitive__header & .hds-dialog-primitive__description
+.hds-dialog-primitive__wrapper-header {
+  flex: none;
+  border-top-left-radius: inherit;
+  border-top-right-radius: inherit;
+  background: var(--token-color-surface-faint);
+}
+
+.hds-dialog-primitive__wrapper-body {
+  flex: 1 1 auto;
+  // this is necessary, otherwise if there are words longer than the width of the available space for the list "item" content
+  overflow-x: hidden;
+  // we want the content to vertically scroll if needed
+  overflow-y: auto;
+  overscroll-behavior: contain;
+  // do we want to move the "border" logic here?
+  // border-top: 1px solid var(--token-color-border-primary);
+}
+
+.hds-dialog-primitive__wrapper-footer {
+  flex: none;
+  // do we want to move the "border" logic here?
+  // border-top: 1px solid var(--token-color-border-primary);
+  border-bottom-right-radius: inherit;
+  border-bottom-left-radius: inherit;
+}
+
+
+// HEADER
+
+.hds-dialog-primitive__header {
+  display: flex;
+  gap: 16px;
+  align-items: flex-start;
+  padding: 16px 24px;
+  color: var(--token-color-foreground-strong); // default color
+  border-radius: inherit;
+}
+
+.hds-dialog-primitive__icon {
+  flex: none;
+  align-self: center;
+}
+
+.hds-dialog-primitive__title {
+  flex-grow: 1;
+}
+
+.hds-dialog-primitive__tagline {
+  margin-bottom: 4px;
+  color: var(--token-color-foreground-faint); // default color
+}
+
+.hds-dialog-primitive__dismiss {
+  align-self: center;
+}
+
+
+// DESCRIPTION
+// (currently used only in Flyout, not in Modal)
+
+.hds-dialog-primitive__description {
+  padding: 0 24px 16px;
+}
+
+
+// BODY
+
+.hds-dialog-primitive__body {
+  padding: 24px;
+  border-top: 1px solid var(--token-color-border-primary);
+}
+
+
+// FOOTER
+
+.hds-dialog-primitive__footer {
+  padding: 16px 24px;
+  background: var(--token-color-surface-faint);
```

Comment:
if we move the background style at wrapper level for the header, we should do the same for the footer